src/core/pointcloud/qgspointcloudlayerprofilegenerator.cpp

```diff
@@ -515,21 +487,64 @@ bool QgsPointCloudLayerProfileGenerator::generateProfile( const QgsProfileGenera
 
   request.setAttributes( attributes );
 
-  switch ( pc->accessType() )
+  mResults = std::make_unique< QgsPointCloudLayerProfileResults >();
+  mResults->copyPropertiesFromGenerator( this );
+  mResults->mMaxErrorInLayerCoordinates = 0;
+
+  for ( auto pc : indexes )
```

Comment:
```suggestion
  for ( QgsPointCloudIndex *pc : std::as_const( indexes ) )
```

Reply:
What does the `std::as_const()` add here? I'm not accessing the raw iterator and I'm getting out copies of (non-const) pointers, so it shouldn't matter, no?

Reply:
It's a qt thing, relating to detachment of the implicitly shared Qt container classes. See https://www.kdab.com/blog-qasconst-and-stdas_const/ for more details.